packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -262,6 +267,26 @@ class UserSessionManager implements IUserSessionManager {
     }
   }
 
+  /**
+   * A function to make an external request to set the cookie from server side
+   * @param key       cookie name
+   * @param value     encrypted cookie value
+   */
+  setServerSideCookie(key: string, value: string): void {
+    let baseUrl = state.lifecycle.activeDataplaneUrl.value;
+    if (typeof state.loadOptions.value.cookieServerUrl === 'string') {
+      baseUrl = state.loadOptions.value.cookieServerUrl;
+    }
+    this.httpClient?.getAsyncData({
+      url: `${baseUrl}/setCookie`,
+      options: {
+        method: 'POST',
+        data: JSON.stringify({ key, value, options: state.storage.cookie.value }),
```

Comment:
Let's use the JSON stringification utility here.

Reply:
You mean stringification with circular?

Reply:
Yes.